spec/integration/indirector/report/yaml.rb

```diff
@@ -0,0 +1,48 @@
+require 'spec_helper'
+
+require 'puppet/transaction/report'
+require 'puppet/indirector/report/yaml'
+require 'puppet/indirector/indirector_testing/report/yaml'
+
+describe Puppet::Transaction::Report::Yaml do
+  describe '#save' do
+    let(:model) { Puppet::IndirectorTesting }
+    let(:indirection) { model.indirection }
+    let(:instance) { model.new('report') }
+    let(:request)  { indirection.request(:report, 'example', instance) }
+    let(:file) { subject.path(request.key) }
+
+    it 'saves the instance of the report as YAML to disk' do
+      subject.save(request)
+      content = File.read(file)
+      expect(content).to match(/value\s*:\s*report/)
```

Comment:
Maybe verify that the content can be parsed as valid YAML, and then assert the value key equals report?